features/admin.connections.v1/components/edit/settings/custom-authenticator-settings.tsx

```diff
@@ -0,0 +1,404 @@
+import { AlertLevels, IdentifiableComponentInterface } from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import { FinalForm, FormRenderProps, FormSpy } from "@wso2is/form";
+import { EmphasizedSegment } from "@wso2is/react-components";
+import { AxiosError } from "axios";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Dispatch } from "redux";
+import { getLocalAuthenticator } from "../../../api/authenticators";
+import {
+    getFederatedAuthenticatorDetails,
+    updateCustomAuthentication,
+    updateFederatedAuthenticator
+} from "../../../api/connections";
+import {
+    AuthenticationPropertiesInterface,
+    ConnectionInterface,
+    CustomAuthConnectionInterface,
+    EndpointAuthenticationType,
+    EndpointAuthenticationUpdateInterface
+} from "../../../models/connection";
+import {
+    handleConnectionUpdateError,
+    handleCustomAuthenticatorUpdateError,
+    handleGetCustomAuthenticatorError
+} from "../../../utils/connection-utils";
+import "./custom-authenticator-settings.scss";
+
+/**
+ * Proptypes for the Custom Local Authenticator edit page component.
+ */
+export interface CustomAuthenticatorSettingsPagePropsInterface extends IdentifiableComponentInterface {
+    /**
+     * Is the authenticator a custom local authenticator.
+     */
+    isCustomLocalAuthenticator: boolean;
+    /**
+     * Is the data still loading.
+     */
+    isLoading?: boolean;
+    /**
+     * Connection details.
+     */
+    connector: CustomAuthConnectionInterface | ConnectionInterface;
+    /**
+     * Callback to update the connector details.
+     */
+    onUpdate: (id: string, tabName?: string) => void;
+}
+
+/**
+ * Custom authenticator settings page.
+ *
+ * @param props - Props injected to the component.
+ * @returns React element.
+ */
+export const CustomAuthenticatorSettings: FunctionComponent<CustomAuthenticatorSettingsPagePropsInterface> = ({
+    isCustomLocalAuthenticator,
+    isLoading,
+    connector,
+    onUpdate,
+    ["data-componentid"]: componentId = "custom-authenticator-settings-page"
+}: CustomAuthenticatorSettingsPagePropsInterface): ReactElement => {
+    const dispatch: Dispatch = useDispatch();
+    const { t } = useTranslation();
+
+    const [ initialValues, setInitialValues ] = useState<EndpointConfigFormPropertyInterface>(null);
+    const [ endpointAuthenticationType, setEndpointAuthenticationType ] = useState<AuthenticationType>(null);
+    const [ isEndpointAuthenticationUpdated, setIsEndpointAuthenticationUpdated ] = useState<boolean>(false);
+
+    /**
+    * This useEffect is utilized only for custom authenticators in order to fetch additional
+    * details related to authenticators.
+    * This is not required for other connections since all the required details
+    * are passed from the parent component.
+    */
```

Comment:
Formatting issue. Please check other places as well.

```suggestion
    /**
     * This useEffect is utilized only for custom authenticators in order to fetch additional
     * details related to authenticators.
     * This is not required for other connections since all the required details
     * are passed from the parent component.
     */
```

Reply:
This will be fixed with a follow up PR